File: tests/fixtures/clean/structural.jl
module CleanFixture

export CreateUser, UpdateUser, normalize_user

struct CreateUser
    id::String
    name::String
    email::String
end

struct UpdateUser
    id::String
    name::String
    email::String
end

const CREATE_SCHEMA = (
    id = String,
    name = String,
    email = String,
)

const UPDATE_SCHEMA = (
    id = String,
    name = String,
    email = String,
)

function normalize_user(user)
    strip(user.name)
end

end

File: tests/fixtures/clean/structural.clj
(ns fixture.clean
  (:require [clojure.string :as str]
            [clojure.set :as set]))

(defrecord CreateUser [id name email])
(defrecord UpdateUser [id name email])

(def schema-a
  {:id string?
   :name string?
   :email string?})

(def schema-b
  {:id string?
   :name string?
   :email string?})

(defn normalize-user [user]
  (update user :name str/trim))
